content/blog/2021/10/02/MonitorTheSolver.adoc

```diff
@@ -0,0 +1,215 @@
+
+== Monitoring the Solver
+
+Now that we know what monitoring is and how it can be useful, let walk through an actual example. We'll be running a modified version of the https://github.com/kiegroup/optaplanner-quickstarts/tree/stable/use-cases/school-timetabling[school timetabling quickstart] with support for multitenancy.
+The complete source code, along with scripts for running, can be https://github.com/Christopher-Chianelli/optaplanner-micrometer-blog[found on GitHub].
+
+== Starting the application
+
+Clone the example code:
+
+[source,bash]
+----
+git clone https://github.com/Christopher-Chianelli/optaplanner-micrometer-blog
+cd optaplanner-micrometer-blog
+----
+
+The application can be started by running development mode:
+
+[source,bash]
+----
+mvn quarkus:dev
+----
+
+Once started, you can access the application at https://localhost:8080. To see available metrics for the application, visit https://localhost:8080/q/metrics.
+
+image:ExampleMonitoringApp.png[Example Application]
+
+== Starting Prometheus
+
+Prometheus has a prebuilt Docker image on Dockerhub that can be used for running Prometheus.
+We'll need to modify it configuration so it'll scrape our metrics endpoint.
+
+Create `prometheus.yml` with the following text:
+
+[source,yaml]
+----
+scrape_configs:
+- job_name: local application
+  scrape_interval: 1s
+  metrics_path: /q/metrics
+  static_configs:
+  - targets:
+    - localhost:8080
+----
+
+This configures Prometheus to scrape metrics from `localhost:8080/q/metrics` every second.
+
+To start Prometheus using the following configuration, run the command below:
+
+[source,bash]
+----
+docker run \
+    --network host \
+    --mount type=bind,source=prometheus.yml,destination=/etc/prometheus/prometheus.yml,ro=true,relabel=shared \
+     prom/prometheus
+----
+
+You can see the Prometheus UI by visiting https://localhost:9090.
+
+image::PrometheusUI.png[Prometheus UI]
+
+== Starting Grafana
+
+https://grafana.com/[Grafana] provides a much more robust UI with additional features.
+Grafana has a prebuilt image on Dockerhub you can use to run Grafana locally. Start it using the command below.
+
+[source,bash]
+----
+docker run --network host grafana/grafana
+----
+
+It may take a while to start. Once started,
+visit https://localhost:3000 to see the Grafana UI. Login with the username "admin" and the password "admin".
+
+== Connecting Grafana to Prometheus
+
+. Click on the Gear icon to go to the Configuration page.
+. Click the "Add data source" button.
+. Select "Prometheus".
+. Enter "http://localhost:9090" for the URL field.
+. (Optional) Set the Scrape Interval to be equal to the one set for Prometheus (1s).
+. Click "Save & Test".
+
+If it is set up correctly, a green textbox should appear above "Save & Test" with text "Data source is working".
+
+image::GrafanaConfigurePrometheus.png[Grafana Prometheus Configuration]
```

Comment:
![image](https://user-images.githubusercontent.com/176880/135223622-e4a0a697-9d09-4ffe-a042-46d595d8a106.png)

Reply:
I'd even use a red text or a red rectangle to annotate in the screenshot what they are looking at: the number of active solvers.